pkg/flow/declare_test.go

```diff
@@ -0,0 +1,308 @@
+package flow_test
+
+import (
+	"context"
+	"os"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/grafana/agent/pkg/flow"
+	"github.com/grafana/agent/pkg/flow/internal/testcomponents"
+	"github.com/grafana/agent/pkg/flow/logging"
+	"github.com/grafana/agent/service"
+	"github.com/stretchr/testify/require"
+)
+
+type testCase struct {
+	name     string
+	config   string
+	expected int
+}
+
+func TestDeclare(t *testing.T) {
```

Comment:
I'd like to see test cases for:

* Ensuring that updating the configuration and changing the definition of a declare block works with the new definition. 

* Ensuring that it's possible to shadow builtin components with custom ones without an error.

Reply:
It will be possible to fully shadow with import. For now, the test can check for shadowing a namespace.

Reply:
something like this?
```
declare "a" {}
declare "b" {
  declare "a" {}
  a {} // no error, use the template defined locally (inside of "b")
}
b {}
```

Reply:
That's another case that should work, so that should be a test as well. But no, I was talking about declaring a component called `prometheus` or `local` or the name of another builtin namespace.

Reply:
Ok, will test both